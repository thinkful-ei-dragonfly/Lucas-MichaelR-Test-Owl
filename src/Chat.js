import React from 'react';
import Event from './Events';
function Chat(props){
  // let events = sort thru the chatEvents array by timestamp;
  // inside of the map if(event.type === 'joined')
  let events = props.events.map((event, i) => <Event key={i} userId={event.participantId} type={event.type} time={event.time} message={event.message} timestamp={event.timestamp} />)
  events.sort(function(a, b) {
    return b.props.timestamp - a.props.timestamp;
  })

  return (
    <div className="chat-events">
      {events}
    </div>
  )
}
export default Chat;
