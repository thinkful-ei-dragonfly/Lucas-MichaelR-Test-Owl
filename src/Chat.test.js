import React from 'react';
import renderer from 'react-test-renderer';

import Chat from './Chat';

import STORE from './store';

describe('<Chat  />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<Chat events={STORE.chatEvents}/>).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/Chat .test.js
    expect(tree).toMatchSnapshot();
  })
})