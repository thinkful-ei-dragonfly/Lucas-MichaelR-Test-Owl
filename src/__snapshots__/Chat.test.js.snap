// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat  /> renders without crashing [snapshot test] 1`] = `
<div
  className="chat-events"
>
  <p>
    <span
      className="chat-user-name"
    >
      Axe Kubicka
    </span>
     joined
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Axe Kubicka
    </span>
     joined the stage
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Ashla Attwool
    </span>
     joined
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Koren Templeton
    </span>
     raise their hand 
    <span
      className="lnr lnr-hand"
    />
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Koren Templeton
    </span>
     clapped 
    <span
      className="lnr lnr-hand"
    />
    <span
      className="lnr lnr-hand"
    />
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Koren Templeton
    </span>
     joined
  </p>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <p>
        <span
          className="chat-user-name"
        >
          Axe Kubicka
        </span>
         
        <span
          className="chat-time"
        >
          4:50 a.m.
        </span>
      </p>
      Hello world
    </div>
  </div>
  <p>
    <span
      className="chat-user-name"
    >
      Caty Flucker
    </span>
     gave a thumbs up 
    <span
      className="lnr lnr-thumbs-down"
    />
  </p>
  <p>
    <span
      className="chat-user-name"
    >
      Frank Runciman
    </span>
     gave a thumbs down 
    <span
      className="lnr lnr-thumbs-up"
    />
  </p>
</div>
`;
